llvm/lib/Transforms/InstCombine/InstructionCombining.cpp

```diff
@@ -2341,9 +2341,12 @@ Value *InstCombiner::getFreelyInvertedImpl(Value *V, bool WillInvertAllUses,
                   !shouldAvoidAbsorbingNotIntoSelect(*cast<SelectInst>(V));
   // Selects/min/max with invertible operands are freely invertible
   if (IsSelect || match(V, m_MaxOrMin(m_Value(A), m_Value(B)))) {
+    bool SaveDoesConsume = DoesConsume;
```

Comment:
I'd personally invert this, and do `bool LocalDoesConsume = DoesConsume;`, pass that as the parameters and then do `DoesConsume = LocalDoesConsume` when it succeeds. I think that's less error-prone than restoring on the error paths.

Reply:
Done.